api/integrations/perforce/zulip_change-commit.py

```diff
@@ -72,13 +72,45 @@
 metadata = git_p4.p4_describe(changelist)  # type: Dict[str, str]
 
 destination = config.commit_notice_destination(changeroot, changelist)  # type: Optional[Dict[str, str]]
+
 if destination is None:
     # Don't forward the notice anywhere
     sys.exit(0)
 
-message = "**{0}** committed revision @{1} to `{2}`.\n\n> {3}".format(
+ignore_missing_stream = None
+try:
+    ignore_missing_stream = config.ZULIP_IGNORE_MISSING_STREAM
+except AttributeError:
+    # old config
+    pass
+
+if ignore_missing_stream:
+    # Check if the destination stream exists yet
+    stream_state = client.get_stream_id(destination["stream"])
+    if stream_state["result"] == "error":
+        # Silently discard the message
+        sys.exit(0)
+
```

Comment:
Agreed - it's clearer (according to the python 2.7 documentation it's implemented by checking it getattr raises an exception)